streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -365,109 +395,200 @@ public void beforeTest() {
         final Properties streamsConfig = streamsConfiguration(
             cache,
             log,
-            storeToTest.name()
+            storeToTest.name(),
+            kind
         );
 
         final StreamsBuilder builder = new StreamsBuilder();
-        if (supplier instanceof KeyValueBytesStoreSupplier) {
-            final Materialized<Integer, Integer, KeyValueStore<Bytes, byte[]>> materialized =
-                Materialized.as((KeyValueBytesStoreSupplier) supplier);
+        if (Objects.equals(kind, "DSL") && supplier instanceof KeyValueBytesStoreSupplier) {
+            setUpKeyValueDSLTopology((KeyValueBytesStoreSupplier) supplier, builder);
```

Comment:
This method was already too long to comfortably read. Doubling its length by adding PAPI was out of the question, so I extracted some methods.